### PR TITLE
Raz/mgmt 1401 clear s3 files

### PR DESCRIPTION
**Pull request by RazRegev**

### CODE REVIEW

**Comment by avishayt** [comment]
I suggest making some const that contains all of the files.  Add a validation to DownloadClusterFiles that param.FileName is one of them, and then also use that const list for deletion.  That way we will know that if a new file is added later it will be deleted.

File: pkg/s3Client/s3Client.go [hunk 1]
```diff
@@ -71,6 +72,20 @@ func (s s3Client) DownloadFileFromS3(ctx context.Context, fileName string, s3Buc
...
 }
 
+func (s s3Client) DeleteFileFromS3(ctx context.Context, fileName string, s3Bucket string) (bool, error) {
+	log := logutil.FromContext(ctx, s.log)
+	log.Infof("Deleting file if %s exists in %s", fileName, s3Bucket)
+	err := s.client.RemoveObject(s3Bucket, fileName)
+	if err != nil {
+		errResponse := minio.ToErrorResponse(err)
+		if errResponse.Code == "NoSuchKey" {
+			return false, nil
```
Comment (avishayt) [inline pkg/s3Client/s3Client.go:82]: This is a good check, but don't care about the bool return value.  We can treat NoSuchKey and deleted as success (return nil), and for any other error return the error.

File: pkg/s3Client/s3Client.go [hunk 2]
```diff
@@ -71,6 +72,20 @@ func (s s3Client) DownloadFileFromS3(ctx context.Context, fileName string, s3Buc
 	return resp, contentLength, nil
 }
 
+func (s s3Client) DeleteFileFromS3(ctx context.Context, fileName string, s3Bucket string) (bool, error) {
+	log := logutil.FromContext(ctx, s.log)
+	log.Infof("Deleting file if %s exists in %s", fileName, s3Bucket)
+	err := s.client.RemoveObject(s3Bucket, fileName)
+	if err != nil {
+		errResponse := minio.ToErrorResponse(err)
+		if errResponse.Code == "NoSuchKey" {
```
Comment (eranco74) [inline pkg/s3Client/s3Client.go:81]: Add a log

File: pkg/s3Client/s3Client.go [hunk 3]
```diff
@@ -71,6 +72,20 @@ func (s s3Client) DownloadFileFromS3(ctx context.Context, fileName string, s3Buc
 	return resp, contentLength, nil
 }
 
+func (s s3Client) DeleteFileFromS3(ctx context.Context, fileName string, s3Bucket string) (bool, error) {
```
Comment (eranco74) [inline pkg/s3Client/s3Client.go:75]: No need for the bool in the return

File: internal/bminventory/inventory.go [hunk 4]
```diff
@@ -1740,6 +1740,10 @@ func (b *bareMetalInventory) ResetCluster(ctx context.Context, params installer.
 		}
 	}
 
+	if err := b.deleteS3ClusterFiles(ctx, &c); err != nil {
+		return installer.NewResetClusterInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
```
Comment (filanov) [inline internal/bminventory/inventory.go:1744]:
```suggestion
return common.NewApiError(http.StatusInternalServerError, err)
```

File: pkg/s3Client/s3Client.go (hunk 3 above)
Comment (filanov) [reply pkg/s3Client/s3Client.go:75]: +1

File: internal/bminventory/inventory.go [hunk 5]
```diff
@@ -1819,6 +1840,15 @@ func (b *bareMetalInventory) ResetCluster(ctx context.Context, params installer.
 	return installer.NewResetClusterAccepted().WithPayload(&c.Cluster)
 }
 
+func (b *bareMetalInventory) deleteS3ClusterFiles(ctx context.Context, c *common.Cluster) error {
+	for _, name := range clusterFileNames {
+		if err := b.s3Client.DeleteFileFromS3(ctx, fmt.Sprintf("%s/%s", c.ID, name), b.S3Bucket); err != nil {
```
Comment (filanov) [inline internal/bminventory/inventory.go:1845]: what will happen if you try to delete file that no longer exists?

**Pull request merged**